mypyc/test-data/run-loops.test

```diff
@@ -571,3 +571,81 @@ print([x for x in native.Vector2(4, -5.2)])
 [out]
 Vector2(x=-2, y=3.1)
 \[4, -5.2]
+
+[case testRunForFilter]
+def f(a: list[int]) -> int:
+    s = 0
+    for x in filter(lambda x: x % 2 == 0, a):
+        s += x
+    return s
+
+[file driver.py]
+from native import f
+print(f([1, 2, 3, 4, 5, 6]))
+print(f([1, 3, 5]))
+print(f([]))
+
+[out]
+12
+0
+0
+
+[case testRunForFilterNone]
+def f(a: list[int]) -> int:
+    c = 0
+    for x in filter(None, a):
+        c += 1
+    return c
+
+[file driver.py]
+from native import f
+print(f([0, 1, 2, 3, 4, 5, 6]))
+
+[out]
+6
+
+[case testRunForFilterNative]
+def f(x: int) -> int:
+    return x % 2
+def g(a: list[int]) -> int:
+    c = 0
+    for x in filter(f, a):
+        c += 1
+    return c
+
+[file driver.py]
+from native import g
+print(g([0, 1, 2, 3, 4, 5, 6]))
+
+[out]
+3
+
+[case testRunForFilterPrimitiveOp]
+def f(a: list[list[int]]) -> int:
+    c = 0
+    for x in filter(len, a):
+        c += 1
+    return c
+
+[file driver.py]
+from native import f
+print(f([[], [0, 1], [], [], [2, 3, 4], [5, 6]]))
+
+[out]
+3
+
+[case testRunForFilterEdgeCases]
+def f(a: list[int]) -> int:
+    s = 0
+    for x in filter(lambda x: x > 10, a):
+        s += x
+    return s
+
+[file driver.py]
+from native import f
+print(f([5, 15, 25]))
+print(f([]))
+
+[out]
```

Comment:
Please merge all the tests under a single `[case ...]`, leave out `driver.py`, use assertions instead of `[out]` and have test cases in `def test_...` functions. (See my similar comment in the `map` PR for the motivation.)